pyomo/util/latex_printer.py

```diff
@@ -0,0 +1,619 @@
+            int: handle_num_node,
+            float: handle_num_node,
+            NegationExpression: handle_negation_node,
+            ProductExpression: handle_product_node,
+            DivisionExpression: handle_division_node,
+            PowExpression: handle_pow_node,
+            AbsExpression: handle_abs_node,
+            UnaryFunctionExpression: handle_unary_node,
+            Expr_ifExpression: handle_exprif_node,
+            EqualityExpression: handle_equality_node,
+            InequalityExpression: handle_inequality_node,
+            RangedExpression: handle_ranged_inequality_node,
+            _GeneralExpressionData: handle_named_expression_node,
+            ScalarExpression: handle_named_expression_node,
+            kernel.expression.expression: handle_named_expression_node,
+            kernel.expression.noclone: handle_named_expression_node,
+            _GeneralObjectiveData: handle_named_expression_node,
+            _GeneralVarData: handle_var_node,
+            ScalarObjective: handle_named_expression_node,
+            kernel.objective.objective: handle_named_expression_node,
+            ExternalFunctionExpression: handle_external_function_node,
+            _PythonCallbackFunctionID: handle_functionID_node,
+            LinearExpression: handle_sum_expression,
+            SumExpression: handle_sum_expression,
+            MonomialTermExpression: handle_monomialTermExpression_node,
+            IndexedVar: handle_var_node,
+            IndexTemplate: handle_indexTemplate_node,
+            Numeric_GetItemExpression: handle_numericGIE_node,
+            TemplateSumExpression: handle_templateSumExpression_node,
+        }
+
+    def exitNode(self, node, data):
+        return self._operator_handles[node.__class__](node, *data)
+
+
+def latex_printer(
+    pyomoElement, filename=None, useAlignEnvironment=False, splitContinuousSets=False
+):
+    '''
+    This function produces a string that can be rendered as LaTeX
+
+    pyomoElement: The thing to be printed to LaTeX.  Accepts Blocks (including models), Constraints, and Expressions
+
+    filename: An optional file to write the LaTeX to.  Default of None produces no file
+
+    useAlignEnvironment: Default behavior uses equation/aligned and produces a single LaTeX Equation (ie, ==False).
+        Setting this input to True will instead use the align environment, and produce equation numbers for each
+        objective and constraint.  Each objective and constraint will be labeled with its name in the pyomo model.
+        This flag is only relevant for Models and Blocks.
+
+    splitContinuous: Default behavior has all sum indices be over "i \in I" or similar.  Setting this flag to
+        True makes the sums go from: \sum_{i=1}^{5} if the set I is continuous and has 5 elements
+
+    '''
+
+    # Various setup things
+    isSingle = False
+    if not isinstance(pyomoElement, _BlockData):
+        if isinstance(pyomoElement, pe.Objective):
```

Comment:
For maintainability, why not do something like this:
```python
objectives = []
constraints = []
expressions = []

if isinstance(pyomo_element, pyo.Objective):
    objectives.append(pyomo_element)
    templatize_fcn = templatize_constraint

... (etc.)
```

Reply:
Corrected to avoid the double if, but avoiding the use of the slow `append` function